ruma-events-macros/src/event_enum.rs

```diff
@@ -265,6 +255,9 @@ fn expand_any_enum_with_deserialize(
 ) -> syn::Result<TokenStream> {
     let attrs = &input.attrs;
     let event_type_str = &input.events;
+
+    // This needs to remove the `Any` since every enum is generated by this method and
+    // they all have an event structs that shares a name with `ident` minus `Any`.
     let event_struct = Ident::new(&ident.to_string().replace("Any", ""), ident.span());
```

Comment:
Basically same concern here, IIRC this was even the main thing that made me want to switch to another approach. It seems like this function doesn't actually use the new `EnumKind` type, why?

Reply:
It has to be able to generate `Stripped*`, `Redacted*`, `*Event`, `*EventContent`, and so on. I'm trying to come up with a way around it, any ideas?

Reply:
I'm not actually that familiar with the code anymore 😆 I will take a look.